### PR TITLE
Add gusd vault

### PR DESCRIPTION
**Pull request by antonnell**
Added GUSD
Modifications to store to better handle exceptions on contract calls (stops perma loading).
Fixed approval calls to accommodate for decimals < 18

### CODE REVIEW

**Comment by milkyklim** [comment]
@x48-crypto need to confirm that strategies are fine first and check\merge this one after that.

File: src/stores/store.jsx
Comment (x48-crypto) [inline src/stores/store.jsx:1346]: Should we be using MAX_UINT256 for approval amount?

```
const MAX_UINT256 = new BigNumber(2)
  .pow(256)
  .minus(1)
  .toString();
```

File: src/components/vault/vault.jsx
Comment (x48-crypto) [inline src/components/vault/vault.jsx:601]: Right now `apyOneWeekSample` will fall back to `apyInceptionSample` if there is not enough data for the specified timeframe. This means that currently `apyOneWeekSample` will be greater than zero once harvest is called. This means that as soon as harvest is called this check will pass, which is probably not the intention.

Some people have asked me to set APY samples to null if there is not enough data for a given timeframe. If I were to do that, that would mean that we would never show vault stats until one week's worth of data has been collected. Thus, inception APY would show as '0.00' until one week's worth of data has been collected.

If the desired behavior is to prevent any vault APY from being shown until one week's worth of data is collected then this change won't work as intended until I update the API to not fall back to inception APY (I can do this soon!)

Comment (x48-crypto) [inline src/components/vault/vault.jsx:601]: Do we need to make this change on `asset.jsx` as well?

**Review by x48-crypto — APPROVED** [review]
Looks good! I tested the UI locally as well (did not test deposit/withdraw).

I noticed strategy is empty for now, but it seems the strategy has not been set yet.

I made a comment that `apyOneWeekSample` may not act as you expect it to.

Also we may want to update the `asset.jsx` and `dashboard.jsx` `_getAPY` functions as well with your new logic.

I think we may want to change token approval amount to `MAX_UINT256` unless you have a specific reason for using a lower approval value.

**Pull request merged**
